File: Level1/소수만들기.py
#소수 만들기
def solution(num):
    answer = 0
    for i in range(len(num)):
        for j in range(i+1,len(num)):
            for k in range(j+1,len(num)):
                tot=num[i]+num[j]+num[k] #3개 선택
                if tot==2: #만약 합이 2면 소수
                    answer+=1
                for x in range(2,tot):
                    if tot%x==0: #소수가 아니므로 tot는 x의 배수
                        break
                    elif x==tot-1: #소수 판별
                        answer+=1
    return answer

File: Level1/소수 만들기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

bool is_primenumber(int n) {
    if (n == 0 || n == 1)
        return false;
    for (int i = 2; i < n / 2; i++) {
        if (n % i == 0)
            return false;
    }
    return true;
}

// nums_len은 배열 nums의 길이입니다.
int solution(int nums[], size_t nums_len) {
    int answer = 0;
    for (int i = 0; i < nums_len; i++) {
        for (int j = i + 1; j < nums_len; j++) {
            for (int k = j + 1; k < nums_len; k++) {
                int tmp_n = nums[i] + nums[j] + nums[k];
                if (is_primenumber(tmp_n))
                    answer++;
            }
        }
    }
    return answer;
}